### PR TITLE
Replace component hashing with sheet hashing

### LINKED ISSUE #288: Update stylesheet runtime to cache based on stylesheets instead of hash
Currently we pass a `hash` to every CS component to use for caching at runtime.

Now with moving to atomic declarations this doesn’t make sense because each declaration may or may not have already been added to the head (or rendered on the server).

- Remove `hash` from ast
- Remove `hash` from style component

Internally in the style component iterate through each sheet (children) - if it hasn’t been added - render it/move to head - else don’t.

### PR DESCRIPTION
**Pull request by madou**
Just smashed this out before going out haha, let me know what you think.

Closes #288

### CODE REVIEW

**Comment by pgmanutd** [comment]
> Just smashed this out before going out haha, let me know what you think.
> 
> Closes #288

🙄 🙄 oooo..you are so fast. Slow down a bit hehehe 😛

**Comment by madou** [comment]
image

Bundle size went up i'll see if i can reduce it next week

**Comment by pgmanutd** [comment]
Looking into it Michael 👀. Give me sometime 😄

File: packages/runtime/src/sheet.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ function createStyleElement(opts: StyleSheetOpts): HTMLStyleElement {
  *
  * @param opts StyleSheetOpts
  */
-export const createStyleSheet = (opts: StyleSheetOpts) => {
+export default function createStyleSheet(opts: StyleSheetOpts) {
```
Comment (pgmanutd) [inline packages/runtime/src/sheet.tsx:25]: nit: hehe. I thought you like named export 😛. Probably if we are using default export, it should match filename right 🤔?

**Review by pgmanutd — APPROVED** [review]
lgtm

File: packages/runtime/src/sheet.tsx (hunk 1 above)
Comment (madou) [reply packages/runtime/src/sheet.tsx:25]: only did this to reduce file size down haha - it reduces it a few bytes

Comment (pgmanutd) [reply packages/runtime/src/sheet.tsx:25]: 😱 wooo, never guessed that. Damn. Nice.

Comment (pgmanutd) [reply packages/runtime/src/sheet.tsx:25]: And btw how it reduced the size 🤔 thinking

Comment (madou) [reply packages/runtime/src/sheet.tsx:25]: when minified default imports reduce down to `a` - something small

named imports cant be reduced it seems (you probably can but requires more complex minification)

Comment (pgmanutd) [reply packages/runtime/src/sheet.tsx:25]: ohh right. hehe. I totally forgot minification here 😛 ahh another drawback of using named export then hehehehe 😛

**Pull request merged**